src/pages/AcceptInvitation/pages/FillRequest/components/FillRequestForm.tsx

```diff
@@ -34,25 +35,19 @@ export default function RequestForm({
     control,
     getErrorMessage,
   } = useForm(
-    {
-      ...credentialFields?.reduce(
-        (acc, field) => ({
-          ...acc,
-          [field.key]: field.value || '',
-        }),
-        {} as Record<string, string>,
-      ),
-    },
+    credentialFields?.reduce((acc, field) => {
+      acc[field.key] = field.value || ''
+      return acc
+    }, {} as Record<string, string>),
     yup =>
       yup.object().shape({
         ...orgGroupRequest.group?.rules?.reduce((acc, rule) => {
-          return rule.required
-            ? {
-                ...acc,
-                [rule.name]: yup.string().email().required(),
-              }
-            : acc
-        }, {}),
+          if (!rule.required) return acc
+
+          acc[rule.name] = yup.string().required()
+
+          return acc
```

Comment:
why not?
```suggestion
          if (rule.required) {
            acc[rule.name] = yup.string().required()
          }

          return acc
```

now you're returning acc twice